pop_tools/grid.py

```diff
@@ -391,3 +391,101 @@ def _compute_corners(ULAT, ULONG):
     corner_lon[0, :, 3] = corner_lon[1, :, 3] - (corner_lon[2, :, 3] - corner_lon[1, :, 3])
 
     return corner_lat, corner_lon
+
+
+@jit(cache=True, nopython=True)
+def dzu_from_dzt(dzt):
+    var = dzt
+    dimz, dimy, dimx = dzt.shape
+    outvar = np.full_like(dzt, fill_value=0)
+
+    for k in np.arange(dimz):
+        for j in np.arange(dimy - 1):
+            for i in np.arange(dimx - 1):
+                outvar[k, j, i] = np.min(
+                    np.array(
+                        [var[k, j, i], var[k, j + 1, i], var[k, j, i + 1], var[k, j + 1, i + 1]]
+                    )
+                )
+
+    return outvar
+
+
+def calc_dzu_dzt(grid):
+    """
+    Calculates DZT from DZ.
+
+    Parameters
+    ----------
+    grid: Dataset
+        An xarray Dataset containing grid variables. This must contain partial bottom
+        cell information: KMT and DZBC.
+
+    Returns
+    -------
+    DZT, DZU: DataArray
+
+    Notes
+    -----
+    From Frank's zulip convo
+    https://zulip.cloud.ucar.edu/#narrow/stream/9-CGD-OCE/topic/pop-tools/near/2864
```

Comment:
It would be great if we could get a summary or similar from these zulip exchanges posted here for posterity. That way the knowledge get lost in private inboxes :)

Reply:
It's the code in the comment :)

Reply:
Thanks , I just saw it 😀!